src/main/scala/strawman/collection/View.scala

```diff
@@ -86,6 +86,20 @@ object View extends IterableFactory[View] {
     def iterator(): Iterator[A] = Iterator.iterate(start)(f).take(len)
     override def knownSize: Int = 0 max len
   }
+  
+  case class Intersperse[A, B >: A](underlying: Iterable[A], sep: B) extends View[B] {
+    def iterator(): Iterator[B] = underlying.iterator().intersperse(sep)
+    override def knownSize: Int = if (underlying.knownSize > 0) (2 * underlying.knownSize - 1) else underlying.knownSize
+  }
+  
+  case class IntersperseAppend[A, B >: A](underlying: Iterable[A], start: B, sep: B, end: B) extends View[B] {
```

Comment:
I don’t think `IntersperseAppend` is a good name. What do you think of `IntersperseAndSurround`, or just `IntersperseSurround`?

Reply:
Also, you can just use a single type parameter `A` everywhere.